src/modules/user/user.usecase.ts

```diff
@@ -82,6 +82,10 @@ export class UserUseCases {
     return this.userRepository.findByUsername(email);
   }
 
+  getUserByEmailAddress(email: UserAttributes['email']) {
```

Comment:
Use `findByUsername`, as multiple users could have the same email on sharedWorkspaces, the username is like the email, but it does not change when a user joins a shared workspace